core/src/blocktree_processor.rs

```diff
@@ -84,13 +94,22 @@ fn execute_batches(batches: &[TransactionBatch]) -> Result<()> {
 /// 2. Process the locked group in parallel
 /// 3. Register the `Tick` if it's available
 /// 4. Update the leader scheduler, goto 1
-pub fn process_entries(bank: &Bank, entries: &[Entry], randomize: bool) -> Result<()> {
+pub fn process_entries(bank: &Arc<Bank>, entries: &[Entry], randomize: bool) -> Result<()> {
+    process_entries_with_callback(bank, entries, randomize, &None)
+}
+
+fn process_entries_with_callback(
+    bank: &Arc<Bank>,
+    entries: &[Entry],
+    randomize: bool,
+    entry_callback: &Option<ProcessCallback>,
```

Comment:
nit:  I usually try to do `Option<&ProcessCallback>` here to avoid the weird looking `&None`

cc: https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref